parma/group/parma_group.cc

```diff
@@ -27,25 +28,41 @@ typedef Parma_GroupCode GroupCode;
 
 static void runInGroups(
     apf::Mesh2* m,
+    pcu::PCU *PCUObj,
     Remap& inMap,
     Remap& groupMap,
     Remap& outMap,
-    GroupCode& code)
+    GroupCode& code
+    )
 {
-  int self = PCU_Comm_Self();
+  int self;
+  pcu::PCU* expandedPCU;
+  if(PCUObj == nullptr){
+    self = m->getPCU()->Self();
+    expandedPCU = m->getPCU();
+  } else {
+    self = PCUObj->Self();
+    expandedPCU = PCUObj;
+  }
+  
   int groupRank = inMap(self);
   int group = groupMap(self);
-  MPI_Comm oldComm = PCU_Get_Comm();
+  
   MPI_Comm groupComm;
-  MPI_Comm_split(oldComm, group, groupRank, &groupComm);
-  PCU_Switch_Comm(groupComm);
-  if (m)
+  MPI_Comm_split(expandedPCU->GetMPIComm(), group, groupRank, &groupComm);
+  auto groupedPCU = std::unique_ptr<pcu::PCU>(new pcu::PCU(groupComm));
+  if (m){
+    m->switchPCU(groupedPCU.get());
```

Comment:
switchPCU function may need review for a possible alternative method.
Also used in:
test/fusion.cc
test/ptnParma.cc
test/split.cc
test/ugrid.cc
test/zsplit.cc